src/main/java/com/avaya/letscode/CartItem.java

```diff
@@ -23,7 +23,7 @@ public int getQuantity() {
         }
 
         // if item has offer, return the quantity based on offer
-        return (this.offer).getAdjustedQuantity(this.product, this.quantity);
+        return (this.offer).getAdjustedQuantity(this);
```

Comment:
We can refactor this in a tertiary operator... 
return (this.offer != null) ? this.offer.getAdjustedQuantity(this) : this.quantity;